.github/workflows/tests.yaml

```
@@ -0,0 +1,127 @@
+name: tests
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  unit-test:
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # ratchet:actions/checkout@v3
+
+      - name: Build coverage-instrumented binary
+        run: |
+          make build-go-cover && sudo make -B install
+
+      - name: Set up Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # ratchet:actions/setup-go@v4
+        with:
+          go-version: '1.21'
+
+      - name: Run Unit-Test
+        run: |
+          mkdir /tmp/unit/
+          go test \
+            -v \
+            -cover ./... \
+            -skip TestHarpoon \
+            -args -test.gocoverdir=/tmp/unit/
+
+      - name: Upload cover profiles
+        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # ratchet:actions/upload-artifact@v3
+        with:
+          name: unit-test
+          path: /tmp/unit/
+
+  integration-test:
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # ratchet:actions/checkout@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@93397bea11091df50f3d7e59dc26a7711a8bcfbe # ratchet:actions/setup-go@v4
+        with:
+          go-version: '1.21'
```

The latest version, maybe?
 
```suggestion
          go-version: '1.22'
```

---

1.21 is the version required to build harpoon, for now we can keep it